nixos/modules/services/web-apps/grist-core.nix

```diff
@@ -0,0 +1,167 @@
+{ config, lib, pkgs, ... }:
+
+let
+  inherit (lib)
+    literalExpression
+    types
+    mkOption
+    mkDefault
+    mkPackageOption
+    mkEnableOption
+    mkIf
+    ;
+
+  cfg = config.services.grist-core;
+
+
+in
+{
+  meta.maintainers = with lib.maintainers; [ soyouzpanda ];
+
+  options.services.grist-core = {
+    package = mkPackageOption pkgs "grist-core" { };
+    enable = mkEnableOption "Grist core";
+
+    user = mkOption {
+      type = types.str;
+      default = "grist-core";
+      description = "User account under which grist-core runs.";
+    };
+
+    group = mkOption {
+      type = types.str;
+      default = "grist-core";
+      description = "Group under which grist-core runs.";
+    };
+
+    pythonEnv = mkOption {
+      internal = true;
+      visible = false;
+      type = types.package;
+      default = pkgs.python3.withPackages (ps: with ps; [
+        astroid
+        asttokens
+        chardet
+        et-xmlfile
+        executing
+        friendly-traceback
+        iso8601
+        lazy-object-proxy
+        openpyxl
+        phonenumbers
+        pure-eval
+        python-dateutil
+        roman
+        six
+        sortedcontainers
+        stack-data
+        typing-extensions
+        unittest-xml-reporting
+        wrapt
+      ]);
+      example = literalExpression ''
+        pkgs.python3.withPackages (ps: with ps; [
+          astroid
+          asttokens
+          chardet
+          et-xmlfile
+          executing
+          friendly-traceback
+          iso8601
+          lazy-object-proxy
+          openpyxl
+          phonenumbers
+          pure-eval
+          python-dateutil
+          roman
+          six
+          sortedcontainers
+          stack-data
+          typing-extensions
+          unittest-xml-reporting
+          wrapt
+        ]);
+      '';
+    };
+
+    settings = mkOption {
+      type = with types; attrsOf (oneOf [ str int bool ]);
```

Comment:
This is invalid, the settings are passed directly to the unit environment which only accepts `str`, `path` or `null`